src/components/panel/base-panel.tsx

```diff
@@ -0,0 +1,28 @@
+import { Layout } from "@lifesg/react-design-system/layout";
+import { ComponentProps } from "react";
+import { useBuilder } from "../../context-providers/builder/hook";
+import { BasePanelHeader } from "./base-panel-header";
+import { BasePanelWrapper } from "./base-panel.styles";
+import { Toolbar } from "./toolbar";
+
+export interface IBasePanelProps extends ComponentProps<"div"> {
```

Comment:
The practice is to have the interfaces specified in a `types.ts` file that sits in `panel` folder. Refer to CONTRIBUTING.md for more info. 

And also don't need to extend `ComponentProps`. Just specify children as a prop
```
children: React.ReactNode;
```